src/Textarea.tsx

```diff
@@ -7,12 +9,32 @@ export interface ITextareaProps extends ITextProps {
   autosize?: boolean;
   minRows?: number;
   maxRows?: number;
-  as?: any;
+
+  value?: string;
+  onChange?: (value: string) => void;
 }
 
-export const Textarea = styled<ITextareaProps>(Text as any).attrs({
-  as: ({ autosize }: ITextareaProps) => (autosize ? AutosizeTextarea : 'textarea'),
-})(
+export const BasicTextArea = (props: ITextareaProps) => {
+  const { autosize, minRows, maxRows, onChange = noop, ...rest } = props;
+
+  const [value, setValue] = React.useState(props.value || '');
+  const internalValue = props.hasOwnProperty('value') ? props.value : value;
```

Comment:
Instead of managing this extra value, might be able to leverage `useEffect` in a clever way, as below. Note, note 100% on the re-render logic, would need to test it out.

```tsx
export const BasicTextArea = (props: ITextareaProps) => {
  const { autosize, minRows, maxRows, onChange = noop, ...rest } = props;

  const [value, setValue] = React.useState(props.value || '');
  
  React.useEffect(() => {
    if (value !== props.value) setValue(props.value);
  }, [props.value])

  const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
    setValue(event.target.value);
    onChange(event.target.value);
  };

  if (autosize) {
    return (
      <AutosizeTextarea {...rest} minRows={minRows} maxRows={maxRows} value={value} onChange={handleChange} />
    );
  }

  return <textarea {...rest} value={value} onChange={handleChange} />;
};
```

If that works, then we can create a sweet little re-usable controllable value hook, as below.

```tsx
export const useControllableValue = (externalValue) => {
  const [value, setValue] = React.useState(externalValue || '');
  
  React.useEffect(() => {
    if (value !== externalValue) setValue(externalValue);
  }, [externalValue])

  return [value, setValue];
}

export const BasicTextArea = (props: ITextareaProps) => {
  const { autosize, minRows, maxRows, onChange = noop, ...rest } = props;

  const [value, setValue] = useControllableValue(props.value);
  
  const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
    setValue(event.target.value);
    onChange(event.target.value);
  };

  if (autosize) {
    return (
      <AutosizeTextarea {...rest} minRows={minRows} maxRows={maxRows} value={value} onChange={handleChange} />
    );
  }

  return <textarea {...rest} value={value} onChange={handleChange} />;
};
```

Reply:
@marbemac @casserni 
extracted that controllableValue here https://github.com/stoplightio/ui-kit/pull/15/commits/b8a7da108ba429138eba78658baa7e778677f6aa
Working flawlessly.

Reply:
@P0lip Solid! Once Code-editor is merged in I will go back and update these components! Merging for now